src/nbl/video/COpenGLCommandBuffer.h

```diff
@@ -542,14 +542,40 @@ class COpenGLCommandBuffer final : public IGPUCommandBuffer
 
     COpenGLCommandBuffer(core::smart_refctd_ptr<const ILogicalDevice>&& dev, E_LEVEL lvl, core::smart_refctd_ptr<IGPUCommandPool>&& _cmdpool, system::logger_opt_smart_ptr&& logger, const COpenGLFeatureMap* _features);
 
+    void releaseResourcesBackToPool()
+    {
+        freeSpaceInCmdPool();
+        m_commands.clear();
+    }
+
+    bool resetCommon() override
+    {
+        // TODO resources for all command buffers already got freed from the command pool,
+        // this function should just make sure the command buffer stops pointing to memory that has
+        // been freed from the pool
+        // (this currently causes a crash due to double free)
+        releaseResourcesBackToPool();
+        return IGPUCommandBuffer::resetCommon();
+    }
+
     inline bool begin(core::bitflag<E_USAGE> _flags) override final
     {
-        reset(E_RESET_FLAGS::ERF_RELEASE_RESOURCES_BIT);
+        checkForCommandPoolReset();
+        // skip reset if command pool was reset above and state is initial
+        if (canReset() && m_state != ES_INITIAL)
+            reset(E_RESET_FLAGS::ERF_RELEASE_RESOURCES_BIT);
+
         return IGPUCommandBuffer::begin(_flags);
```

Comment:
hoist this stuff to `IGPUCommandBuffer::begin` and just leave an empty `virtual void begin_impl` for vulkan to override

Reply:
Resolved.